File: frontend/src/api/models/AttemptResponse.ts
export type GameStatus = 'NOT_STARTED' | 'IN_PROGRESS' | 'WON' | 'LOST'

export type AttemptResponse = {
  guess: string
  feedback: string // "CCPAA" format: C=correct, P=present, A=absent
  attemptNumber: number
  status: GameStatus
}

File: application/src/main/kotlin/com/doctolib/kotlinwordletraining/dto/AttemptResponse.kt
package com.doctolib.kotlinwordletraining.dto

import com.doctolib.kotlinwordletraining.entity.GameStatus

data class AttemptResponse(
    val guess: String = "",
    val feedback: String = "",
    val attemptNumber: Int = 0,
    val status: GameStatus = GameStatus.NOT_STARTED,
)
